test/e2e/specs/crossTabs.js

```diff
@@ -0,0 +1,80 @@
+import assert from 'assert';
+import TestApp from '../pageobjects/TestApp';
+import { openPKCE } from '../util/appUtils';
+import { loginPopup } from '../util/loginUtils';
+
+const openMultipleTabs = async () => {
+  await openPKCE();
+  // open in new tab
+  await browser.newWindow('/');
+  await openPKCE();
+};
+
+const assertSameTokensInTabs = async (tabTokenMap, handles) => {
+  let preToken;
+  for (let i = 0; i < handles.length; i++) {
+    const handle = handles[i];
+    await browser.switchToWindow(handle);
+    tabTokenMap[handle] = {
+      idToken: await TestApp.idToken.then(el => el.getText()),
+      accessToken: await TestApp.accessToken.then(el => el.getText())
+    };
+    if (preToken) {
+      assert(JSON.stringify(preToken) === JSON.stringify(tabTokenMap[handle]));
+    }
```

Comment:
should there be an `else {` here? We only need/want to set `preToken` once, right?

Reply:
should not really matter, any unequal will fail the test.